### PR TITLE
Use `dayjs` in `TimeInput`

### PR DESCRIPTION
**Pull request by kulyk**
Migrates the `TimeInput` to `dayjs` instead of `moment`

Turns out `dayjs` doesn't handle time-only formats out of the box (like `HH:mm`) and [it needs the `CustomParsePlugin` to be installed](https://day.js.org/docs/en/parse/string-format). I had to bring back app-wide `days` initialization for that (@ranquild added it in #34802, but then reworked in #34897)

**To verify:** CI should be green 🟢

### CODE REVIEW

File: frontend/src/metabase/app.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ import "number-to-locale-string";
 // Should be imported before any other metabase import
 import "ee-overrides"; // eslint-disable-line import/no-duplicates
 
+import "metabase/lib/dayjs";
```
Comment (uladzimirdev) [inline frontend/src/metabase/app.js:14]: I think @ranquild had a very similar PR which he abandoned

Comment (uladzimirdev) [reply frontend/src/metabase/app.js:14]: @kulyk check this https://github.com/metabase/metabase/pull/34897/files#r1366806961 , initial PR is https://github.com/metabase/metabase/pull/34802/

Comment (kulyk) [reply frontend/src/metabase/app.js:14]: Yeah, I wasn't sure what he meant by "hard to manage plugins loaded globally" 🤔 
I'm seeing 5 failed E2E test groups and I'm hoping they're flakes :lol_sob:

Comment (ranquild) [reply frontend/src/metabase/app.js:14]: It broke storybook, so lets import this file also there

Comment (ranquild) [reply frontend/src/metabase/app.js:14]: @kulyk if you decided to use this approach please move other dayjs.extend to this file

**Review by ranquild — APPROVED** [review]
Please take a look at comments

File: frontend/src/metabase/app.js (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/app.js:14]: My initial idea for reverting the PR was that we can get away from using plugins globally, but if such basic functionality requires a plugin I guess that's not realistic

**Comment by kulyk** [comment]
@ranquild thanks! Please take a look if the last two commits make sense

**Comment by ranquild** [comment]
@kulyk `localeData` is not needed

**Review by ranquild — APPROVED** [review]

**Pull request merged**